From 11c8680c8d0ec0e9c834f57efdfd3827453f4c6a Mon Sep 17 00:00:00 2001
Message-Id: <11c8680c8d0ec0e9c834f57efdfd3827453f4c6a.1652042168.git.tommyhebb@gmail.com>
From: Thomas Hebb <tommyhebb@gmail.com>
Date: Sun, 8 May 2022 11:46:08 -0700
Subject: [PATCH 1/2] Remove Debian-specific bits from Makefiles

The main change here is to get rid of architecture-specific library
directories.
---
 Common.mk                              |  4 ++--
 Library.mk                             | 22 +++++++---------------
 USBDM_API_Example/Common.mk            |  4 ++--
 USBDM_Programmer_API_Example/Common.mk |  4 ++--
 4 files changed, 13 insertions(+), 21 deletions(-)

diff --git a/Common.mk b/Common.mk
index c941f787..aa94a528 100755
--- a/Common.mk
+++ b/Common.mk
@@ -53,7 +53,7 @@ SHARED_LIBDIR := ../Shared/$(MULTIARCH)
 
 #===========================================================
 # Where to find private libraries on linux
-PKG_LIBDIR="/usr/lib/$(MULTIARCH)/${PKG_NAME}"
+PKG_LIBDIR="/usr/lib/${PKG_NAME}"
 
 #===========================================================
 # Where to build
@@ -257,7 +257,7 @@ ifeq ($(UNAME_S),Windows)
    JAVA_INC := -I$(JAVA_DIR)/include
    JAVA_INC += -I$(JAVA_DIR)/include/win32
 else
-   JAVA_INC := -I/usr/lib/jvm/default-java/include -I/usr/lib/jvm/default-java/include/linux $(jvm_includes)
+   JAVA_INC := -I/usr/lib/jvm/default/include -I/usr/lib/jvm/default/include/linux $(jvm_includes)
 endif
 
 #=============================================================
diff --git a/Library.mk b/Library.mk
index 00e780a3..76b8047d 100644
--- a/Library.mk
+++ b/Library.mk
@@ -10,8 +10,7 @@ USBDM_VERSION := $(MAJOR_VERSION).$(MINOR_VERSION)
 
 #===========================================================
 # Where to find private libraries on linux
-USBDM_LIBDIR32="/usr/lib/i386-linux-gnu/usbdm"
-USBDM_LIBDIR64="/usr/lib/x86_64-linux-gnu/usbdm"
+USBDM_LIBDIR="/usr/lib/usbdm"
 
 #===========================================================
 # Shared directories - Relative to child directory
@@ -40,16 +39,8 @@ else
    # Assume Linux
    DIRS = $(COMMON_DIRS)
    BITNESS ?= $(shell getconf LONG_BIT)
-   ifeq ($(BITNESS),32)
-      TARGET_BINDIR   ?= ../PackageFiles/bin/i386-linux-gnu
-      TARGET_LIBDIR   ?= ../PackageFiles/lib/i386-linux-gnu
-      BUILDDIR_SUFFIX ?= .i386
-   endif
-   ifeq ($(BITNESS),64)
-      TARGET_BINDIR   ?= ../PackageFiles/bin/x86_64-linux-gnu
-      TARGET_LIBDIR   ?= ../PackageFiles/lib/x86_64-linux-gnu
-      BUILDDIR_SUFFIX ?= .x86_64
-   endif
+   TARGET_BINDIR   ?= ../PackageFiles/bin
+   TARGET_LIBDIR   ?= ../PackageFiles/lib
 endif
 
 ifeq ($(UNAME_S),Windows)
@@ -248,7 +239,7 @@ ifeq ($(UNAME_S),Windows)
    JAVA_INC := -I$(PROGRAM_DIR_JAVA)/Java/jdk1.8.0_31/include
    JAVA_INC += -I$(PROGRAM_DIR_JAVA)/Java/jdk1.8.0_31/include/win32
 else
-   JAVA_INC := -I/usr/lib/jvm/default-java/include
+   JAVA_INC := -I/usr/lib/jvm/default/include
 endif
 
 #=============================================================
@@ -315,12 +306,13 @@ endif
 ifneq ($(OS),Windows_NT)
    ifeq ($(BITNESS),32)
       CFLAGS  += -m32
-      LDFLAGS += -m32 -Wl,-rpath,${USBDM_LIBDIR32}
+      LDFLAGS += -m32
    endif
    ifeq ($(BITNESS),64)
       CFLAGS  += -m64
-      LDFLAGS += -m64 -Wl,-rpath,${USBDM_LIBDIR64}
+      LDFLAGS += -m64
    endif
+   LDFLAGS += -Wl,-rpath,${USBDM_LIBDIR}
 endif
 
 CFLAGS  += ${THREADS} -Wall -shared ${GCC_VISIBILITY_DEFS}
diff --git a/USBDM_API_Example/Common.mk b/USBDM_API_Example/Common.mk
index 5f79c882..bc877eab 100755
--- a/USBDM_API_Example/Common.mk
+++ b/USBDM_API_Example/Common.mk
@@ -49,7 +49,7 @@ SHARED_LIBDIR := ../Shared/$(MULTIARCH)
 
 #===========================================================
 # Where to find private libraries on linux
-PKG_LIBDIR="/usr/lib/$(MULTIARCH)/${PKG_NAME}"
+PKG_LIBDIR="/usr/lib/${PKG_NAME}"
 
 #===========================================================
 # Where to build
@@ -258,7 +258,7 @@ ifeq ($(UNAME_S),Windows)
    JAVA_INC := -I$(JAVA_DIR)/include
    JAVA_INC += -I$(JAVA_DIR)/include/win32
 else
-   JAVA_INC := -I/usr/lib/jvm/default-java/include -I/usr/lib/jvm/default-java/include/linux $(jvm_includes)
+   JAVA_INC := -I/usr/lib/jvm/default/include -I/usr/lib/jvm/default/include/linux $(jvm_includes)
 endif
 
 #=============================================================
diff --git a/USBDM_Programmer_API_Example/Common.mk b/USBDM_Programmer_API_Example/Common.mk
index 5f79c882..bc877eab 100755
--- a/USBDM_Programmer_API_Example/Common.mk
+++ b/USBDM_Programmer_API_Example/Common.mk
@@ -49,7 +49,7 @@ SHARED_LIBDIR := ../Shared/$(MULTIARCH)
 
 #===========================================================
 # Where to find private libraries on linux
-PKG_LIBDIR="/usr/lib/$(MULTIARCH)/${PKG_NAME}"
+PKG_LIBDIR="/usr/lib/${PKG_NAME}"
 
 #===========================================================
 # Where to build
@@ -258,7 +258,7 @@ ifeq ($(UNAME_S),Windows)
    JAVA_INC := -I$(JAVA_DIR)/include
    JAVA_INC += -I$(JAVA_DIR)/include/win32
 else
-   JAVA_INC := -I/usr/lib/jvm/default-java/include -I/usr/lib/jvm/default-java/include/linux $(jvm_includes)
+   JAVA_INC := -I/usr/lib/jvm/default/include -I/usr/lib/jvm/default/include/linux $(jvm_includes)
 endif
 
 #=============================================================
-- 
2.36.0

